sema4ai/vscode-client/src/actionServer.ts

```diff
@@ -294,3 +296,125 @@ export const startActionServer = async (directory: Uri) => {
     actionServerTerminal.sendText(`cd ${directory.fsPath}`);
     actionServerTerminal.sendText(`${location} start --port=${port}`);
 };
+
+const askForAccessCredentials = async (): Promise<string | undefined> => {
+    const access_credentials: string = await window.showInputBox({
+        "placeHolder": "Control Room Access Credentials",
+        "prompt": "Please provide the Control Room access credentials.",
+        "ignoreFocusOut": true,
+        "password": true,
+        "validateInput": (access_credentials: string): string | Thenable<string> => {
+            const regex = /^\d{1,}:.{11,}$/;
+            if (!regex.test(access_credentials)) {
+                return "Invalid access credentials: use form 1234:xxx...x";
+            }
+        },
+    });
+
+    return access_credentials;
+};
+
+export const verifyLogin = async (actionServerLocation: string): Promise<ActionServerVerifyLoginOutput | undefined> => {
+    const result: ActionResult<ActionServerVerifyLoginOutput> = await commands.executeCommand(
+        roboCommands.SEMA4AI_ACTION_SERVER_CLOUD_VERIFY_LOGIN_INTERNAL,
+        {
+            action_server_location: actionServerLocation,
+        }
+    );
+
+    if (!result.success) {
+        window.showErrorMessage(`Failed to verify login: ${result.message}`);
+        return;
+    }
+
+    return result.result;
+};
+
+const askUserForHostname = async (
+    actionServerLocation: string,
+    progress: Progress<{ message?: string; increment?: number }>
+): Promise<string | undefined> => {
+    let defaultHostname = "https://us1.robocorp.com";
+
+    progress.report({ message: "Getting default hostname" });
+    const loginOutput = await verifyLogin(actionServerLocation);
+    if (loginOutput && loginOutput.logged_in) {
+        defaultHostname = loginOutput.hostname;
+    }
+
+    progress.report({ message: "Input hostname" });
+    const hostname: string = await window.showInputBox({
```

Comment:
This should probably be gotten from the `rcc` profile right? (we can integrate as is for now, just wanted to double-check if that's the case).

Reply:
Most probably this should come from rcc. 
We should create a Linear Task for this & add it to the backlog, just so we don't forget.

Reply:
Well currently there isn't anything we can get from RCC. I have been asking, and the answer has been that there isn't anything yet and we should just use the us1.robocorp.com. But basically there is no default, and will not be, the user has to set it. So in the future most probably there shouldn't be any defaults. But again need decision on this...